src/lib/search.js

```diff
@@ -74,20 +75,39 @@ exports.distinctVals = function(valsIn) {
     var vals = valsIn.slice();  // otherwise we sort the original array...
     vals.sort(exports.sorterAsc);
 
-    var l = vals.length - 1;
-    var minDiff = (vals[l] - vals[0]) || 1;
-    var errDiff = minDiff / (l || 1) / 10000;
-    var v2 = [vals[0]];
+    var first;
+    for(first = 0; first < vals.length; first++) {
+        if(vals[first] !== BADNUM) break;
+    }
```

Comment:
Oh interesting, I wouldn't have guessed that the problem traced back to `distinctVals`, nice find!

TIL:
> all undefined elements are sorted to the end of the array, with no call to `compareFunction`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

So perhaps all we need is the loop to find `last` and the rest of this can stay as it was, as we'll never have `undefined` anywhere else after a `sort`?

Reply:
maybe worth testing this in IE before relying on it. I've confirmed in Chrome, FF, Safari on my mac so seems robust on modern browsers.

If it were to fail on IE, I wouldn't trust that sorting works at all then with interspersed `undefined` values, and better would be to filter these out before even starting (instead of the `slice()` call)

Reply:
@alexcjohnson good catch! Not working on IE11.

Reply:
@alexcjohnson Correction. It appear to be working on IE11.
![IE11](https://user-images.githubusercontent.com/33888540/81240324-1ae14b80-8fd5-11ea-8c97-c66cd0b2904f.png)

Reply:
@alexcjohnson correction number 2: It is sometimes working and sometime not.
So let's fix that for IE11.

Reply:
OK. The logic is modified by https://github.com/plotly/plotly.js/pull/4814/commits/2f3b098f78180ecb36ed36f27b96c35c13352d96 & tested in IE11 and it works.